engine/src/build/config/apple/apple_sdk.gni

```diff
@@ -2,16 +2,168 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//build/toolchain/rbe.gni")
+
+assert(host_os == "mac")
+
+# Common args across all Apple platforms.
 declare_args() {
-  # Path to the Xcode toolchain (XcodeDefault.xctoolchain).
-  apple_host_toolchain_path = ""
+  # The MACOSX_DEPLOYMENT_TARGET variable used when compiling.
+  # Must be of the form x.x.x for Info.plist files.
+  mac_deployment_target = ""
+
+  # Path to the host Xcode toolchain (XcodeDefault.xctoolchain).
+  mac_host_toolchain_path = ""
+
+  # Path to the macOS platform.
+  mac_platform_path = ""
+
+  # Path to a specific version of the macOS SDK, not including a backslash at
+  # the end.
+  mac_sdk_path = ""
+}
+
+# Must be set above or by gn.
+assert(mac_deployment_target != "")
+
+if (is_ios) {
+  # iOS-specific args.
+  declare_args() {
+    # Path to the iOS platform to use.
+    #
+    # When empty, this will use the default platform based on the value of
+    # use_ios_simulator.
+    ios_platform_path = ""
+
+    # Path to the iOS SDK to use.
+    #
+    # When empty this will use the default SDK based on the value of
+    # use_ios_simulator.
+    ios_sdk_path = ""
+
+    # Set to true when targeting a simulator build on iOS. False means that the
+    # target is for running on the device. The default value is to use the
+    # Simulator except when targeting GYP's Xcode builds (for compat with the
```

Comment:
what's GYP?

Reply:
Ha -- wow that's a bit dated. `gyp` was the build system that came before `gn`.

See: https://gyp.gsrc.io